### ISSUE TITLE
Pyright doesn't detect enum members when they are not uppercase

### ISSUE THREAD

**Issue by jlricon** [microsoft/pyright#852]
**Describe the bug**
Python enums allow lowercase member names, but Pyright is not interpreting them correctly, leading to the issue below:

**To Reproduce**
```python
from enum import Enum
from typing import NoReturn


class SomeClass(Enum):
  A = 1
  B = 2
def assert_never(x: NoReturn) -> NoReturn:
    """Used to cause Mypy to catch impossible cases."""
    # https://github.com/python/mypy/issues/6366#issuecomment-560369716
    assert False, "Unhandled type: {}".format(type(x).__name__)

def test(SomeClass):
  if x == SomeClass.A:
    pass
  elif x == SomeClass.B:
    pass
  else:
    assert_never(x)
```
This works fine, however if the members of the enum are changed to lowercase a/b, pyright thinks the enum has not been exhaustively matched and raises an error for assert_never

**Expected behavior**
All members of the enum, regardless of capitalization should be treated equally.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of Pyright and Pylance.

**Comment by jlricon** [comment]
Thanks for th swift response!

**Comment by erictraut** [comment]
This is now fixed in Pyright 1.1.54, which I just published. It will also be included in the next release of Pylance.
